File: kattis/problems/character/character.cpp
#include <iostream>

using namespace std;

long long fact( long long n )
{
  if ( n == 0 ) {
    return 1;
  }
  
  return n * fact( n - 1 );
}

int main()
{
  int n;
  cin >> n;
  
  long long result = 0;
  long long nFact = fact( n );
  for ( long long i = 2; i <= n; i++ ) {
    result += nFact / ( fact( i ) * fact( n - i ) );
  }
  
  cout << result << endl;
}

File: kattis/problems/character/character.java
import java.math.BigInteger;
import java.util.Scanner;

public class character {
  static BigInteger fact( BigInteger n )
  {
    if ( n.equals(0) ) {
      return BigInteger.valueOf(1);
    }
    
    return n.multiply( fact( n.subtract(BigInteger.valueOf(1)) ));
  }

  public static void main(String[] args) {
    BigInteger n;

    Scanner sc = new Scanner( System.in );
    int input = sc.nextInt();
    n = BigInteger.valueOf( input );
    BigInteger nFact = fact( n );
    BigInteger result = BigInteger.valueOf(0);
    for ( int i = 2; i <= input; i++ ) {
      result = result.add( nFact.divide( fact( BigInteger.valueOf( i ) ).multiply( fact( n.subtract(BigInteger.valueOf(i))) )) );
    }
    
    System.out.println(result);
  }
}
